micro_http/src/common/mod.rs

```diff
@@ -28,6 +28,32 @@ pub enum RequestError {
     InvalidRequest,
 }
 
+/// Errors associated with a HTTP Connection.
+#[derive(Debug)]
+pub enum ConnectionError {
+    /// The request parsing has failed.
+    ParseError(RequestError),
+    /// Could not perform a stream operation successfully.
+    StreamError(std::io::Error),
+    /// Attempted to read or write on a closed connection.
+    ConnectionClosed,
+    /// Attempted to write on a stream when there was nothing to write.
+    BadWriteAttempt,
+}
+
+impl PartialEq for ConnectionError {
+    fn eq(&self, other: &Self) -> bool {
+        use self::ConnectionError::*;
+        match (self, other) {
```

Comment:
Sorry, I didn't notice this earlier.

I will assume the PartialEq implementation is actually only used in tests. Nevertheless, since it is defined here and can be used externally, we should make sure it is accurate.
To that extend you will need to also compare:
- The RequestError inside ParseError (this should be easy as RequestError derives PartialEq)
- io::Error in StreamError (io::Error does not implement PartialEq, but you can "trick it" by comparing err.to_string() == other_err.to_string())

Reply:
Actually I had suggested George to not worry about comparing the internal errors as I thought these are only used in tests where that is irrelevant.

@georgepisaltu check if internal error is relevant in tests comparing these errors and if not you can move this `PartialEq` implementation under `mod tests` and keep it like this. Otherwise, implement comparisons for the internal/contained objects as well.